src/pages/workspaces/workspace/Notebooks.js

```diff
@@ -68,119 +69,130 @@ const noNotebooksMessage = div({ style: { fontSize: 20 } }, [
-          disabled: !canWrite,
-          tooltip: !canWrite && noWrite,
-          tooltipSide: 'left',
-          onClick: () => onDelete()
-        }, [makeMenuIcon('trash'), 'Delete'])
-      ])
-    }, [
-      h(Link, { 'aria-label': 'Notebook menu', onClick: e => e.preventDefault() }, [
-        icon('cardMenuIcon', {
-          size: listView ? 18 : 24
-        })
-      ])
+        }
+      }, [makeMenuIcon('copy-to-clipboard'), 'Copy notebook URL to clipboard']),
+      h(MenuButton, {
+        disabled: !canWrite,
+        tooltip: !canWrite && noWrite,
+        tooltipSide: 'left',
+        onClick: () => onRename()
+      }, [makeMenuIcon('renameIcon'), 'Rename']),
+      h(MenuButton, {
+        disabled: !canWrite,
+        tooltip: !canWrite && noWrite,
+        tooltipSide: 'left',
+        onClick: () => onDelete()
+      }, [makeMenuIcon('trash'), 'Delete'])
+    ])
+  }, [
+    h(Link, { 'aria-label': 'Notebook menu', onClick: e => e.preventDefault() }, [
+      icon('cardMenuIcon', {
+        size: listView ? 18 : 24
+      })
     ])
+  ])
 
-    const title = div({
-      title: printName(name),
-      style: _.merge({
-        ...Style.elements.card.title, whiteSpace: 'normal', overflowY: 'auto'
-      }, listView ? {
-        marginLeft: '1rem'
-      } : { height: 60, padding: '1rem' })
-    }, printName(name))
+  const title = div({
+    title: printName(name),
+    style: _.merge({
+      ...Style.elements.card.title, whiteSpace: 'normal', overflowY: 'auto'
+    }, listView ? {
+      marginLeft: '1rem'
+    } : { height: 60, padding: '1rem' })
+  }, printName(name))
 
-    return a({
-      href: notebookLink,
-      style: {
-        ...Style.elements.card.container,
-        ...notebookCardCommonStyles(listView),
-        flexShrink: 0
-      }
-    }, listView ? [
-      notebookMenu,
+  return a({
+    href: notebookLink,
+    style: {
+      ...Style.elements.card.container,
+      ...notebookCardCommonStyles(listView),
+      flexShrink: 0
+    }
+  }, listView ? [
+    notebookMenu,
+    title,
+    div({ style: { flexGrow: 1 } }),
+    locked ? h(Clickable, { style: { display: 'flex', paddingRight: '1rem', color: colors.dark(0.75) }, tooltip: `This notebook is currently being edited by ${lockedBy || 'another user'}` }, [icon('lock')]) : undefined,
+    h(TooltipTrigger, { content: Utils.makeCompleteDate(updated) }, [
+      div({ style: { fontSize: '0.8rem', marginRight: '0.5rem' } },
+        `Last edited: ${Utils.makePrettyDate(updated)}`)
+    ])
+  ] : [
+    div({ style: { display: 'flex' } }, [
       title,
       div({ style: { flexGrow: 1 } }),
-      isRecent ? div({ style: { display: 'flex', color: colors.warning(), marginRight: '2rem' } }, 'Recently Edited') : undefined,
+      locked ? h(Clickable, { style: { display: 'flex', padding: '1rem', color: colors.dark(0.75) }, tooltip: `This notebook is currently being edited by ${lockedBy || 'another user'}` }, [icon('lock')]) : undefined
```

Comment:
These might want to be deduped at some point, but for the moment same change as above:
```suggestion
      locked && h(Clickable, { style: { display: 'flex', padding: '1rem', color: colors.dark(0.75) }, tooltip: `This notebook is currently being edited by ${lockedBy || 'another user'}` }, [icon('lock')])
```